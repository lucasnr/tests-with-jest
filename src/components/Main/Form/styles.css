#main .form {
	display: flex;
	flex-direction: column;
	padding: 2rem;
	padding-bottom: 0px;
	width: 100%;
}

#main .form > *:not(:last-child) {
	margin-bottom: 1.5rem;
}

#main .form input,
#main .form textarea {
	background-color: #fff;
	border: none;
	font-size: 0.875rem;
	padding: 0.625rem 1.125rem;
	width: 100%;
}

#main .form input {
	border-radius: 9999px;
}

#main .form textarea {
	border-radius: 0.5rem;
}

#main .form button {
	background-color: #d44;
	border: 0.0625rem solid #d44;
	border-radius: 9999px;
	color: #fff;
	font-size: 0.875rem;
	padding: 0.625rem 1.125rem;
	text-align: center;
	transition: background-color 0.2s ease, color 0.2s ease;
}

#main .form button:hover,
#main .form button:focus {
	background: none;
	color: #d44;
}

@media (min-width: 768px) {
	#main .form {
		align-items: flex-start;
	}
}
