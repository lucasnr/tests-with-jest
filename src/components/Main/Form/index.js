import React, { useCallback, useState } from 'react';

import './styles.css';

function Form({ onSubmit }) {
	const [title, setTitle] = useState();
	const [text, setText] = useState();
	const handleSubmit = useCallback(
		(event) => {
			event.preventDefault();

			const data = {
				title,
				text,
			};

			onSubmit(data);
		},
		[onSubmit, text, title]
	);

	return (
		<form className="form" onSubmit={handleSubmit}>
			<input
				name="title"
				type="text"
				placeholder="Enter the title"
				required
				value={title || ''}
				onChange={(event) => setTitle(event.target.value)}
			/>
			<textarea
				name="text"
				cols="30"
				rows="10"
				placeholder="Enter the text"
				required
				value={text || ''}
				onChange={(event) => setText(event.target.value)}
			/>

			<button type="submit">Add new card</button>
		</form>
	);
}

export default Form;
