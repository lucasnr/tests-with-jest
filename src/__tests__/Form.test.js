import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from '../components/Main/Form';

describe('Form', () => {
	it('renders without crashing and matches snapshot', () => {
		const wrapper = shallow(<Form />);
		expect(wrapper).toMatchSnapshot();
	});

	it('calls function on submit', () => {
		const mockFn = jest.fn();
		const wrapper = mount(<Form onSubmit={mockFn} />);

		wrapper.simulate('submit');
		expect(mockFn).toHaveBeenCalled();

		wrapper.simulate('submit');
		expect(mockFn).toHaveBeenCalledTimes(2);
	});
});
