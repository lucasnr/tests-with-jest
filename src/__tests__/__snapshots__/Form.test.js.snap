// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders without crashing and matches snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Enter the title"
    required={true}
    type="text"
    value=""
  />
  <textarea
    cols="30"
    name="text"
    onChange={[Function]}
    placeholder="Enter the text"
    required={true}
    rows="10"
    value=""
  />
  <button
    type="submit"
  >
    Add new card
  </button>
</form>
`;
